gbpservice/nfp/config_orchestrator/agent/otc_service_events.py

```diff
@@ -54,3 +69,65 @@ def _delete_service(self, context, resource):
                                                ctxt, resource,
                                                "DELETE",
                                                network_function_event=True)
+
+    @core_pt.poll_event_desc(event='SERVICE_CREATE_PENDING', spacing=5)
+    def create_sevice_pending_event(self, ev):
+        event_data = ev.data
+        ctxt = n_context.Context.from_dict(event_data['context'])
+
+        if (event_data['service_type']).lower() == 'firewall':
+            request_data = self.fw_agent._prepare_request_data(
+                                                    ctxt,
+                                                    event_data[
+                                                        'nf_id'],
+                                                    event_data[
+                                                        'resource_id'],
+                                                    event_data[
+                                                        'fw_mac'],
+                                                    event_data[
+                                                        'service_type'])
+            msg = ("%s : %s " % (request_data, event_data['fw_mac']))
+            LOG(LOGGER, 'INFO', '%s' % (msg))
+
+        if (event_data['service_type']).lower() == 'loadbalancer':
+            request_data = self.lb_agent._prepare_request_data(
+                                                ctxt,
+                                                event_data[
+                                                    'nf_id'],
+                                                event_data[
+                                                    'resource_id'],
+                                                event_data[
+                                                    'vip_id'],
+                                                event_data[
+                                                    'service_type'])
+            msg = ("%s : %s " % (request_data, event_data['vip_id']))
+            LOG(LOGGER, 'INFO', '%s' % (msg))
+
+        if (event_data['service_type']).lower() == 'vpn':
+            request_data = self.vpn_agent._prepare_request_data(
+                                                ctxt,
+                                                event_data[
+                                                     'nf_id'],
+                                                event_data[
+                                                     'resource_id'],
+                                                event_data[
+                                                     'ipsec_id'],
+                                                event_data[
+                                                     'service_type'])
+            msg = ("%s : %s " % (request_data, event_data['ipsec_id']))
+            LOG(LOGGER, 'INFO', '%s' % (msg))
+
+        if request_data['nf']['status'] == 'ACTIVE':
```

Comment:
adding try catch block so that it will handle the None cases for all the attributes. and stop the polling event if None occurs.